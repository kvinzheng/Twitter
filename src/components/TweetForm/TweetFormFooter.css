.TweetFormFooter {
  justify-content: space-between;
  flex-wrap: wrap;
  position: relative;
}

.TweetFormFooter-warning {
  color: red;
  position: absolute;
  right: 5px;
  font-size: 12px;
}

.TweetFormFooter-item {
  cursor: pointer;
  background-color: transparent;
  height: 42px;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  text-decoration: none;
  line-height: 16px;
  white-space: nowrap;
  font-weight: 700;
  transition: all 0.15s linear 0s;
  margin: 0px;
  font-size: 13px;
  display: inline-flex;
  -webkit-box-align: center;
  align-items: center;
  -webkit-box-pack: center;
  justify-content: center;
  mix-blend-mode: multiply;
  min-width: 40px;
  min-height: 40px;
  padding: 8px 12px;
  border-radius: 50px;
  color: rgb(54, 65, 65);
}

.TweetFormFooter-item a {
  color: black;
  text-decoration: none;
}
.TweetFormFooter-item:hover {
  background-color: #d3d3d3;
}

.TweetFormFooter-remain-count {
  float: right;
}
