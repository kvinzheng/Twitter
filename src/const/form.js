export const TEXT_AREA_COUNT = 280;
export const FETCH_DELAY_TIME = 1000;