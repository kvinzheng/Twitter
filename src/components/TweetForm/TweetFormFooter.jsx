import React from "react";
import {  Target, Camera } from "react-feather";

import "./TweetFormFooter.css";

const TweetFormFooter = ({error, countRemain}) => {
  return (
    <div className="TweetFormFooter-options">
      {error.length ? (
        <div className="TweetFormFooter-warning">{error}</div>
      ) : null}

      <div className="TweetFormFooter-item">
        <Target size="20" />
      </div>
      <div className="TweetFormFooter-item">
        <Camera size="20" />
      </div>
      <div className="TweetFormFooter-item TweetFormFooter-remain-count">{countRemain}</div>
    </div>
  );
};
export default TweetFormFooter;
