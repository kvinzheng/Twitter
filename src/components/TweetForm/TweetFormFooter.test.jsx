import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import TweetFormFooter  from "./TweetFormFooter.jsx";
import { TEXT_AREA_COUNT } from "../../const/form";


configure({ adapter: new Adapter() });

describe("TweetFormFooter Component", () => {
  it("should render without any data defined", () => {
    const component = shallow(<TweetFormFooter error={""} status={TEXT_AREA_COUNT} />);
    expect(component).toMatchSnapshot();
  });

  it("should render with props passed in", () => {
    const component = shallow(
      <TweetFormFooter
        error={
          "please editing an @ followed by 2 alphanumeric characters (a-z, 0-9)"
        }
        status={TEXT_AREA_COUNT}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("Renders with correct classes", () => {
    const component = shallow(
      <TweetFormFooter
        error={
          "please editing an @ followed by 2 alphanumeric characters (a-z, 0-9)"
        }
        status={TEXT_AREA_COUNT}
      />
    );
    expect(component.find(".TweetFormFooter").length).toBe(1);
    expect(component.find(".TweetFormFooter-warning").length).toBe(1);
    expect(component.find(".TweetFormFooter-item").length).toBe(3);
  });
});
