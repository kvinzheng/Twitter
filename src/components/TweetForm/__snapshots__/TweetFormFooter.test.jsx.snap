// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetFormFooter Component should render with props passed in 1`] = `ShallowWrapper {}`;

exports[`TweetFormFooter Component should render without any data defined 1`] = `ShallowWrapper {}`;
